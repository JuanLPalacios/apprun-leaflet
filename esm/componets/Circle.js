import { Circle as LeafletCircle } from 'leaflet';
import { Container } from '../abstracts/Container';
export class Circle extends Container {
    createLayer(props) {
        const { position } = props;
        return new LeafletCircle(position, props).addTo(props.context);
    }
    updateLayer(circle, props, prevProps) {
        if (props.position != null && props.position !== prevProps.position) {
            circle.setLatLng(props.position);
        }
        if (props.radius != null && props.radius !== prevProps.radius) {
            circle.setRadius(props.radius);
        }
        circle.setStyle(Object.assign(Object.assign({}, prevProps), props));
    }
}
//# sourceMappingURL=Circle.js.map