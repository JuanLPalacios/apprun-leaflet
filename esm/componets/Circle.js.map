{"version":3,"file":"Circle.js","sourceRoot":"","sources":["../../src/componets/Circle.tsx"],"names":[],"mappings":"AAAA,OAAO,EAIL,MAAM,IAAI,aAAa,EAExB,MAAM,SAAS,CAAC;AAKjB,OAAO,EAAE,SAAS,EAAkB,MAAM,wBAAwB,CAAC;AAOnE,MAAM,OAAO,MAAO,SAAQ,SAAuD;IACjF,WAAW,CAAC,KAAqC;QAC/C,MAAM,EAAE,QAAQ,EAAE,GAAG,KAAK,CAAC;QAC3B,OAAO,IAAI,aAAa,CAAC,QAAQ,EAAE,KAAK,CAAC,CAAC,KAAK,CAAC,KAAK,CAAC,OAAO,CAAC,CAAC;IACjE,CAAC;IACD,WAAW,CAAC,MAA0B,EAAE,KAAqC,EAAE,SAAyC;QACtH,IAAI,KAAK,CAAC,QAAQ,IAAI,IAAI,IAAI,KAAK,CAAC,QAAQ,KAAK,SAAS,CAAC,QAAQ,EAAE;YACnE,MAAM,CAAC,SAAS,CAAC,KAAK,CAAC,QAAQ,CAAC,CAAC;SAClC;QACD,IAAI,KAAK,CAAC,MAAM,IAAI,IAAI,IAAI,KAAK,CAAC,MAAM,KAAK,SAAS,CAAC,MAAM,EAAE;YAC7D,MAAM,CAAC,SAAS,CAAC,KAAK,CAAC,MAAM,CAAC,CAAC;SAChC;QACD,MAAM,CAAC,QAAQ,iCAAK,SAAS,GAAK,KAAK,EAAE,CAAC;IAC5C,CAAC;CACF"}