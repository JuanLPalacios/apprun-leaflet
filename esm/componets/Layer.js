import app, { Component } from 'apprun';
import { Layer as LeafletLayer } from 'leaflet';
export class Layer extends Component {
    constructor(props, ...p) {
        super(props, ...p);
        this.view = (state = this.state) => {
            if (!this.layer) {
                this.layer = this.createLayer(state);
            }
            if (state.context) {
                this.layer.addTo(state.context);
            }
            return app.createElement("div", null, state.children);
        };
        this.mounted = (props, children, state) => {
            var _a;
            const context = this.layer;
            state.children = (_a = state.children) === null || _a === void 0 ? void 0 : _a.map(node => typeof node === 'string' ?
                node : (node.tag.prototype instanceof Component ? Object.assign(Object.assign({}, node), { props: Object.assign(Object.assign({}, node.props), { context }) }) : node));
            if (props.context !== state.context) {
                if (state.context) {
                    this.layer.remove();
                }
                this.layer.addTo(props.context);
            }
            this.updateLayer(this.layer, props, state);
            return Object.assign(Object.assign({}, props), { children });
        };
        this.createLayer = (props) => {
            return new LeafletLayer(props);
        };
        // eslint-disable-next-line @typescript-eslint/no-empty-function, @typescript-eslint/no-unused-vars
        this.updateLayer = (layer, props, prevProps) => { };
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
    }
}
//# sourceMappingURL=Layer.js.map