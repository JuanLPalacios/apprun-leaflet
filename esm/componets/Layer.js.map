{"version":3,"file":"Layer.js","sourceRoot":"","sources":["../../src/componets/Layer.tsx"],"names":[],"mappings":"AAAA,OAAO,EAAE,KAAK,IAAI,YAAY,EAAe,MAAM,SAAS,CAAC;AAE7D,OAAO,EAAE,SAAS,EAAkB,MAAM,wBAAwB,CAAC;AAMnE,MAAM,OAAO,KAAM,SAAQ,SAAmC;IAC5D,WAAW,CAAC,KAA+B;QACzC,OAAO,IAAI,YAAY,CAAC,KAAK,CAAC,CAAC;IACjC,CAAC;IAED,WAAW,CAAC,KAAmB,EAAE,KAA+B,EAAE,SAAmC;IACrG,CAAC;CACF"}