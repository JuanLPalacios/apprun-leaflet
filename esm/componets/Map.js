import { Map as LeafletMap, } from 'leaflet';
import { app, Component } from 'apprun';
export class Map extends Component {
    constructor() {
        super(...arguments);
        this.view = (state = this.state) => {
            const context = this.map;
            const { bounds, boundsOptions, center, children, class: _class, className, id, placeholder, style, whenReady, zoom } = state;
            const contents = context ? (children === null || children === void 0 ? void 0 : children.map(node => typeof node === 'string' ?
                node :
                (node.tag.prototype instanceof Component ? Object.assign(Object.assign({}, node), { props: Object.assign(Object.assign({}, node.props), { context }) }) : node))) : (placeholder !== null && placeholder !== void 0 ? placeholder : null);
            const node = this.conainer;
            return app.createElement(app.Fragment, null,
                node || app.createElement("div", Object.assign({}, { style, id, className, class: _class }, { ref: (node) => {
                        if (node) {
                            this.conainer = node;
                        }
                        const context = this.map;
                        if (node !== null && !context) {
                            const map = new LeafletMap(node, state);
                            if (state.eventHandlers)
                                map.on(state.eventHandlers);
                            map.on('popupopen', () => {
                                var _a, _b;
                                return (_b = (_a = map._popup) === null || _a === void 0 ? void 0 : _a._closeButton) === null || _b === void 0 ? void 0 : _b.addEventListener('click', (event) => {
                                    event.preventDefault();
                                });
                            });
                            if (center != null && zoom != null) {
                                map.setView(center, zoom);
                            }
                            else if (bounds != null) {
                                map.fitBounds(bounds, boundsOptions);
                            }
                            if (whenReady != null) {
                                map.whenReady(whenReady);
                            }
                            this.run('setContext', map);
                        }
                    } })),
                app.createElement("div", { style: { display: 'none' } }, contents));
        };
        this.update = {
            'setContext': (state, map) => {
                this.map = map;
                return state;
            }
        };
        this.mounted = (props, children, state) => {
            const map = this.map;
            if (props.center != null && props.center !== state.center) {
                map.setView(props.center);
            }
            if (props.zoom != null && props.zoom !== state.zoom) {
                map.setZoom(props.zoom);
            }
            if (props.bounds != null && props.bounds !== state.bounds) {
                map.fitBounds(props.bounds);
            }
            if (props.whenReady != null) {
                map.whenReady(props.whenReady);
            }
            return Object.assign(Object.assign(Object.assign({}, state), props), { children });
        };
        this.unload = (state) => {
            delete state.center;
            delete state.zoom;
            delete state.bounds;
            this.setState(Object.assign({}, state));
        };
    }
}
//# sourceMappingURL=Map.js.map