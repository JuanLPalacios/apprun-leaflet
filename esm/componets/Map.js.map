{"version":3,"file":"Map.js","sourceRoot":"","sources":["../../src/componets/Map.tsx"],"names":[],"mappings":";;;;;;;;;;;AAAA,OAAO,EACL,GAAG,IAAI,UAAU,GAIlB,MAAM,SAAS,CAAC;AACjB,OAAO,EAAE,GAAG,EAAE,SAAS,EAAS,MAAM,QAAQ,CAAC;AAa/C,MAAM,OAAO,GAAI,SAAQ,SAA4B;IAArD;;QAIE,SAAI,GAAG,CAAC,KAAK,GAAG,IAAI,CAAC,KAAK,EAAE,EAAE;YAC5B,MAAM,OAAO,GAAG,IAAI,CAAC,GAAG,CAAC;YACzB,MAAM,EACJ,MAAM,EACN,aAAa,EACb,MAAM,EACN,QAAQ,EACR,SAAS,EACT,EAAE,EACF,WAAW,EACX,KAAK,EACL,SAAS,EACT,IAAI,KAEF,KAAK,EADJ,OAAO,UACR,KAAK,EAZH,iHAYL,CAAQ,CAAC;YACV,MAAM,QAAQ,GAAG,OAAO,CAAC,CAAC,CAAC,CACzB,QAAQ,aAAR,QAAQ,uBAAR,QAAQ,CAAE,GAAG,CAAC,IAAI,CAAA,EAAE,CAClB,OAAO,IAAI,KAAK,QAAQ,CAAA,CAAC;gBACvB,IAAI,CAAC,CAAC;gBACN,CACG,IAAI,CAAC,GAAW,CAAC,SAAS,YAAY,SAAS,CAAA,CAAC,iCAAM,IAAI,KAAE,KAAK,kCAAM,IAAI,CAAC,KAAK,KAAE,OAAO,OAAK,CAAC,CAAC,IAAI,CACvG,CACJ,CACF,CAAC,CAAC,CAAC,CACF,WAAW,aAAX,WAAW,cAAX,WAAW,GAAI,IAAI,CACpB,CAAC;YACF,MAAM,IAAI,GAAG,IAAI,CAAC,QAAQ,CAAC;YAC3B,OAAO;gBACJ,IAAI,IAAI,2CAAS,EAAE,KAAK,EAAE,EAAE,EAAE,SAAS,EAAE,IAAE,GAAG,EAAE,CAAC,IAA2B,EAAE,EAAE;wBAC/E,IAAG,IAAI,EAAE;4BAAE,IAAI,CAAC,QAAQ,GAAG,IAAI,CAAC;yBAAE;wBAClC,MAAM,OAAO,GAAG,IAAI,CAAC,GAAG,CAAC;wBACzB,IAAI,IAAI,KAAK,IAAI,IAAI,CAAC,OAAO,EAAE;4BAC7B,MAAM,EAAE,KAAK,EAAE,EAAE,EAAE,SAAS,EAAE,GAAG,IAAI,CAAC;4BACtC,MAAM,GAAG,GAAG,IAAI,UAAU,CAAC,IAAI,EAAE,OAAO,CAAC,CAAC;4BAC1C,iEAAiE;4BACjE,kCAAkC;4BAClC,IAAI,MAAM,IAAI,IAAI,IAAI,IAAI,IAAI,IAAI,EAAE;gCAClC,GAAG,CAAC,OAAO,CAAC,MAAM,EAAE,IAAI,CAAC,CAAC;gCAC1B,kCAAkC;6BACnC;iCAAM,IAAI,MAAM,IAAI,IAAI,EAAE;gCACzB,GAAG,CAAC,SAAS,CAAC,MAAM,EAAE,aAAa,CAAC,CAAC;6BACtC;4BACD,kCAAkC;4BAClC,IAAI,SAAS,IAAI,IAAI,EAAE;gCACrB,GAAG,CAAC,SAAS,CAAC,SAAS,CAAC,CAAC;6BAC1B;4BACD,IAAI,CAAC,GAAG,CAAC,YAAY,EAAE,GAAG,EAAE,EAAE,KAAK,EAAE,EAAE,EAAE,SAAS,EAAE,CAAC,CAAC;yBACvD;oBACH,CAAC,IAAG;gBACH,QAAQ,CACL,CAAC;QACT,CAAC,CAAC;QAEF,WAAM,GAAG;YACP,YAAY,EAAE,CAAC,KAAK,EAAE,GAAG,EAAE,EAAE;gBAC3B,IAAI,CAAC,GAAG,GAAG,GAAG,CAAC;gBACf,OAAO,KAAK,CAAC;YACf,CAAC;SACF,CAAC;QAEF,YAAO,GAAG,CAAC,KAAwB,EAAE,QAAe,EAAE,KAAwB,EAAE,EAAE;YAChF,qDAAY,KAAK,GAAK,KAAK,KAAE,QAAQ,IAAG;QAC1C,CAAC,CAAC;IACJ,CAAC;CAAA"}