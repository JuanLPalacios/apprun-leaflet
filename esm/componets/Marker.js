import { Marker as LeafletMarker, } from 'leaflet';
import { Container } from '../abstracts/Container';
export class Marker extends Container {
    createLayer(props) {
        const { position } = props;
        return new LeafletMarker(position, props).addTo(props.context);
    }
    updateLayer(marker, props, prevProps) {
        if (props.context !== prevProps.context) {
            if (prevProps.context) {
                marker.remove();
            }
            marker.addTo(props.context);
        }
        if (props.icon != null && props.icon !== prevProps.icon) {
            marker.setIcon(props.icon);
        }
        if (props.zIndexOffset != null &&
            props.zIndexOffset !== prevProps.zIndexOffset) {
            marker.setZIndexOffset(props.zIndexOffset);
        }
        if (props.opacity != null && props.opacity !== prevProps.opacity) {
            marker.setOpacity(props.opacity);
        }
        if (marker.dragging != null && props.draggable !== prevProps.draggable) {
            if (props.draggable === true) {
                marker.dragging.enable();
            }
            else {
                marker.dragging.disable();
            }
        }
    }
}
//# sourceMappingURL=Marker.js.map