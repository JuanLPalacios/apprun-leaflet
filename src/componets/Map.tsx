import {
  Map as LeafletMap,
  FitBoundsOptions,
  LatLngBoundsExpression,
  MapOptions,
} from 'leaflet';
import { app ,Component, VNode } from 'apprun';

export interface MapContainerProps extends MapOptions {
  bounds?: LatLngBoundsExpression
  boundsOptions?: FitBoundsOptions
  children?: Array<VNode | string>
  className?: string
  id?: string
  placeholder?: VNode | string
  style?: any
  whenReady?: () => void
}

export class Map extends Component<MapContainerProps> {
  map:LeafletMap;
  conainer:HTMLDivElement;

  view = (state = this.state) => {
    const context = this.map;
    const {
      bounds,
      boundsOptions,
      center,
      children,
      className,
      id,
      placeholder,
      style,
      whenReady,
      zoom,
      ...options
    } = state;
    const contents = context ? (
      children?.map(node=>
        typeof node === 'string'?
          node :
          (
            (node.tag as any).prototype instanceof Component? { ...node, props:{ ...node.props ,context } } : node
          )
      )
    ) : (
      placeholder ?? null
    );
    const node = this.conainer;
    return <div>
      {node || <div {...{ style, id, className }} ref={(node: HTMLDivElement | null) => {
        if(node) { this.conainer = node; }
        const context = this.map;
        if (node !== null && !context) {
          const { style, id, className } = node;
          const map = new LeafletMap(node, options);
          if (center != null && zoom != null) {
            map.setView(center, zoom);
          } else if (bounds != null) {
            map.fitBounds(bounds, boundsOptions);
          }
          if (whenReady != null) {
            map.whenReady(whenReady);
          }
          this.run('setContext', map, { style, id, className });
        }
      }}/>}
      {contents}
    </div>;
  };

  update = {
    'setContext': (state, map) => {
      this.map = map;
      return state;
    }
  };

  mounted = (props: MapContainerProps, children: any[], state: MapContainerProps) => {
    const map = this.map;
    if (props.center != null && props.center !== state.center) {
      map.setView(props.center);
    }
    if (props.zoom != null && props.zoom !== state.zoom) {
      map.setZoom(props.zoom);
    }
    if (props.bounds != null && props.bounds !== state.bounds) {
      map.fitBounds(props.bounds);
    }
    if (props.whenReady != null) {
      map.whenReady(props.whenReady);
    }
    return { ...state, ...props, children };
  };
  
  unload = (state: MapContainerProps) => {
    delete state.center;
    delete state.zoom;
    delete state.bounds;
    this.setState({...state});
  };
}