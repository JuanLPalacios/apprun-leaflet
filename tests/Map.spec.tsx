import jsdom from 'jsdom-global';
import app from 'apprun';
import {Map} from '../src/componets/Map';

document = window.document;

const render = (vdom) => app.start(document.body, {}, () => vdom);

describe(Map, () => {
  it('renders', (done) => {
    render(<Map
      center={[51.505, -0.09]}
      zoom={13} scrollWheelZoom={false}
      className="map-cont"
      style={{ height:'400px', width:'400px' }}
    />);
    //const vdom = map['view'](state);
    setTimeout(() => {
      expect(document.body.innerHTML).toMatchSnapshot();
      done();
    }, 1000)
  })

  it('binds click events', (done) => {
    const handler = jest.fn(() => null)
    render(<Map
      style={{ height:'400px', width:'400px' }}
      eventHandlers={{ click: handler }}
    />);
    //const vdom = map['view'](state);
    setTimeout(() => {
      const tile:any = document.querySelector('.leaflet-tile');
      tile.fireEvent("onclick")
      tile.click();
      setTimeout(() => {
        expect(handler).toHaveBeenCalled();
        done();
      }, 1000)
    }, 1000)
  })
});