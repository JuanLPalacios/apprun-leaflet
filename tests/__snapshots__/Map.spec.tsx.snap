// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map renders 1`] = `"<section _c=\\"1666500052654\\" style=\\"height: 400px; width: 400px;\\" class=\\"map-cont\\"><div style=\\"height: 400px; width: 400px; position: relative;\\" class=\\"map-cont leaflet-container leaflet-touch leaflet-grab leaflet-touch-drag leaflet-touch-zoom\\" tabindex=\\"0\\"><div class=\\"leaflet-pane leaflet-map-pane\\" style=\\"left: 0px; top: 0px;\\"><div class=\\"leaflet-pane leaflet-tile-pane\\"></div><div class=\\"leaflet-pane leaflet-overlay-pane\\"></div><div class=\\"leaflet-pane leaflet-shadow-pane\\"></div><div class=\\"leaflet-pane leaflet-marker-pane\\"></div><div class=\\"leaflet-pane leaflet-tooltip-pane\\"></div><div class=\\"leaflet-pane leaflet-popup-pane\\"></div></div><div class=\\"leaflet-control-container\\"><div class=\\"leaflet-top leaflet-left\\"><div class=\\"leaflet-control-zoom leaflet-bar leaflet-control\\"><a class=\\"leaflet-control-zoom-in\\" href=\\"#\\" title=\\"Zoom in\\" role=\\"button\\" aria-label=\\"Zoom in\\" aria-disabled=\\"false\\"><span aria-hidden=\\"true\\">+</span></a><a class=\\"leaflet-control-zoom-out\\" href=\\"#\\" title=\\"Zoom out\\" role=\\"button\\" aria-label=\\"Zoom out\\" aria-disabled=\\"false\\"><span aria-hidden=\\"true\\">âˆ’</span></a></div></div><div class=\\"leaflet-top leaflet-right\\"></div><div class=\\"leaflet-bottom leaflet-left\\"></div><div class=\\"leaflet-bottom leaflet-right\\"><div class=\\"leaflet-control-attribution leaflet-control\\"><a href=\\"https://leafletjs.com\\" title=\\"A JavaScript library for interactive maps\\">Leaflet</a></div></div></div></div><div style=\\"display: none;\\"></div></section>"`;
